### DISCUSSION TITLE
Data被覆寫掉

### DISCUSSION THREAD

**Discussion by pouree** [bol-edu/HLS-SOC-Discussions#199] [Q&A]
各位大神 想請問一下我在寫入不同的資料時，data會被覆寫掉，請問有人也有遇到這個問題嗎?  在麻煩各位解答 感謝

**Comment by yushuodev** [comment]
可能因為BRAM的address只有12個bit，解碼的時候需要注意，資料會不會被覆寫
可以看昨天報告的影片有提到

**Reply by Yanlin900414** [reply]
我們也是遇到這問題，也是這樣子解出來的

**Comment by boledulab** [comment]
目前的 SDRAM 設計 （sdr.v）中，並沒有把 row address 放進去。所以 實際可用的memory 大小，比 BRAM size 要小，可能是造成覆蓋的原因。今天會與 TA 討論後，再跟大家做說明。

**Comment by boledulab** [comment]
Refer to the attached.  TA will update the Lab-SDRAM design.  It shall give you a better idea the address flow from user address to sdram device. 
[sdram-bank.pptx](https://github.com/bol-edu/HLS-SOC-Discussions/files/13867565/sdram-bank.pptx)

**Comment by boledulab** [comment]
Github, lab-sdram 已更新！
